File: CliniCorpapp/src/app/consulta/models/consulta.ts
import { Medico } from "./medico";
import { Paciente } from "./paciente";

export interface Consulta {
    id: number,
    paciente: Paciente,
    medico: Medico,
    dataConsulta: Date,
    descricaoConsulta: string,
    MedicoEspecialista: number,
    statusConsulta: number,
    nome:string,
    especializacao:number,
    nomePaciente: string,
    status: string
}

File: api/CliniCorp.Business/Models/Consulta.cs
using System.ComponentModel.DataAnnotations;

namespace ProjetoDemo
{
    public class Consulta
    {
        [Key]
        public int Id { get; set; }
        public DateTime DataConsulta { get; set; }
        public string DescricaoConsulta { get; set; }
        public int Status { get; set; }
        public string StatusConsulta { get; set; }

        public Medico? Medico { get; set; }
        public Paciente? Paciente { get; set; }

    }
}
